scripts/pylib/twister/twisterlib.py

```diff
@@ -2767,6 +2770,49 @@ def process(self, pipeline, done, message, lock, results):
             else:
                 self.cleanup_artifacts()
 
+    def amend_test_instance(self):
+        lst_file = os.path.join(self.build_dir, "zephyr", "zephyr.lst")
+        if os.path.isfile(lst_file):
+            logger.debug(f"zephyr.lst located: {lst_file}")
+        else:
+            # no zephyr.lst file, cannot do symbol-based amending
+            logger.debug(f"zephyr.lst NOT located: {lst_file}")
+            return
+
+        yaml_testsuite_name = self.instance.name.split(os.path.sep)[-1]
+        logger.debug(f"amending yaml test suite: {yaml_testsuite_name}")
+        intended_testcases = []
+
+        with open(lst_file, 'r') as fp:
+            lst_content = fp.read()
+            logger.debug(f"test instance {self.instance.name} has {len(self.instance.testcases)} cases.")
+            for testcase in self.instance.testcases:
+                # identifier is "yaml_testsuite.test_function" or just "yaml_testsuite"
+                identifier = testcase.name
+                if identifier == yaml_testsuite_name:
+                    # Sometimes twsiter add yaml test suite name to a test instance as a test case.
+                    # Such test case has no real test function.
+                    # Skip it to make stats right
+                    logger.debug(f"skipping: {identifier}")
+                    continue
+
+                test_func_name1 = f"test_{identifier.split('.')[-1]}"
+                test_func_name2 = f"{identifier.split('.')[-1]}"
```

Comment:
> That's because I'm afraid someone may not follow the function name convention. So just in case. :)

This shouldn't take place. We should enforce following the conventions and not adding "sloppiness". And in fact we already are:
https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/pylib/twister/twisterlib.py#L2005